cpp/ql/src/experimental/Security/CWE/CWE-193/ConstantSizeArrayOffByOne.ql

```diff
@@ -79,23 +79,28 @@ predicate isInvalidPointerDerefSink2(DataFlow::Node sink, Instruction i, string
 }
 
 predicate pointerArithOverflow(
-  PointerArithmeticInstruction pai, Field f, int size, int bound, int delta
+  PointerArithmeticInstruction pai, Variable v, int size, int bound, int delta
 ) {
-  pai.getElementSize() = f.getUnspecifiedType().(ArrayType).getBaseType().getSize() and
-  f.getUnspecifiedType().(ArrayType).getArraySize() = size and
+  pai.getElementSize() = v.getUnspecifiedType().(ArrayType).getBaseType().getSize() and
+  v.getUnspecifiedType().(ArrayType).getArraySize() = size and
   semBounded(getSemanticExpr(pai.getRight()), any(SemZeroBound b), bound, true, _) and
   delta = bound - size
 }
 
-module FieldAddressToDerefConfig implements DataFlow::StateConfigSig {
+module ArrayAddressToDerefConfig implements DataFlow::StateConfigSig {
   newtype FlowState =
-    additional TArray(Field f) or
+    additional TArray(Variable v) or
```

Comment:
As I mentioned in https://github.com/github/codeql/pull/13045, we can probably restrict these variables to be the ones satisfying `pointerArithOverflow(_, v, _, _, _)`.